### PR TITLE
feat: add support for Twilio Email

### PR DESCRIPTION
**Pull request by childish-sambino**

### CODE REVIEW

File: lib/sendgrid/base_interface.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  #   - +request_headers+ -> any headers that you want to be globally applied
+  #   - +version+ -> the version of the API you wish to access,
+  #                  currently only "v3" is supported
+  #   - +impersonate_subuser+ -> the subuser to impersonate, will be passed
+  #                              in the "On-Behalf-Of" header
+  #
+  def initialize(auth:, host:, request_headers: nil, version: nil, impersonate_subuser: nil)
+    @auth = auth
+    @host = host
+    @version = version ? version : 'v3'
```
Comment (eshanholtz) [inline lib/sendgrid/base_interface.rb:20]: why even have this field is the only supported value is "v3"?

Comment (childish-sambino) [reply lib/sendgrid/base_interface.rb:20]: v2 still exists so it would just be `api` is guess: https://sendgrid.com/docs/API_Reference/Web_API/mail.html

Comment (eshanholtz) [reply lib/sendgrid/base_interface.rb:20]: Not sure I understand. v2 still exists but our helper lib doesn't support it. According to the docs above, `currently only "v3" is supported`. If that's the case, why do we allow the user to pass in their own value?

Comment (childish-sambino) [reply lib/sendgrid/base_interface.rb:20]: The underlying client doesn't _not_ support it. It's just there are no higher-level helpers to work with v2, but you could still use this client.

Comment (eshanholtz) [reply lib/sendgrid/base_interface.rb:20]: Ah. Makes sense.

**Pull request merged**